server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java

```diff
@@ -17,6 +17,10 @@ public record HealthIndicatorResult(String name, String component, HealthStatus
     implements
         ToXContentObject {
 
+    public static HealthIndicatorResult of(String name, String component, HealthStatus status, String summary) {
+        return new HealthIndicatorResult(name, component, status, summary, (builder, params) -> builder.startObject().endObject());
```

Comment:
I wonder if we should use `HealthIndicatorDetails.EMPTY` here? 
May be it is worth to update that to have empty object inside.

Reply:
I tried that, but the health tests started with JSON parsing errors. Let me dig a bit to find out what's the root cause for that.